.github/workflows/trunk.yaml

```diff
@@ -0,0 +1,35 @@
+---
+name: trunk
+
+permissions: read-all
+"on": [pull_request]
+
+jobs:
+  trunk:
+    concurrency:
+      # yamllint disable-line
+      group: trunk # You can specify a group name for concurrency control
+      # yamllint disable-line
+      cancel-in-progress: true # This will cancel any in-progress runs in the same group
+      # yamllint disable-line
+    runs-on: ubuntu-latest # Specify the OS for the job
+
+    steps:
+      - name: Checkout code
+        # yamllint disable-line
+        uses: actions/checkout@v4 # Ensure you check out your code
+        with:
+          ref: main
+      - name: Install Trunk
+        run: curl https://get.trunk.io -fsSL | bash
+      - name: temp
+        run: echo "abc "
+      - name: temp 2
+        run: cat .trunk/trunk.yaml
+      - name: Initialize trunk
+        run: '[ ! -d ".trunk" ] && trunk init ||
+          echo "Trunk is already initialized"'
```

Comment:
Seems like some of these steps are for debugging, please remove them if not needed

Reply:
No, these are needed.